### PR TITLE
feat(docs): automatically generate `longhornctl` documents from the commandline usage

### PR DESCRIPTION
**Pull request by derekbit**
#### Which issue(s) this PR fixes:

Issue https://github.com/longhorn/longhorn/issues/8992

#### What this PR does / why we need it:

Automatically generate longhornctl documentation from command line usage.

Benefits:
- No need to update README.md with changes in command line usage.
- Users can quickly find usage information without downloading longhornctl and running longhornctl --help.

TODO:
- We need to add `Example` in `cobra.Command{}`. 
- Add a new page in longhorn official document for introducing the CLI and add link to the README.md.

### CODE REVIEW

**Review by c3y1huang — COMMENTED** [review]
This is a great PR for automating the doc generation. Thank you!

Additionally, it would be nice if we could have @longhorn/doc improve the menu descriptions in a separate issue.

File: cmd/remote/longhornctl.go [hunk 1]
```diff
@@ -73,9 +74,32 @@ func newCmdLonghornctl() *cobra.Command {
...
 	cmd.AddCommand(subcmd.NewCmdGlobalOptions())
+	cmd.AddCommand(newCmdDocs())
 
 	filters := []string{"options"}
 	templates.ActsAsRootCommand(cmd, filters, groups...)
 
 	return cmd
 }
+
+func newCmdDocs() *cobra.Command {
```
[the remark this answers is not in the document]
Comment (derekbit) [reply cmd/remote/longhornctl.go]: Updated. Thank you.

File: cmd/remote/longhornctl.go [hunk 2]
```diff
@@ -73,9 +74,32 @@ func newCmdLonghornctl() *cobra.Command {
...
+	return &cobra.Command{
+		Use:   "docs [output directory]",
+		Short: "Generate markdown documentation for the CLI",
+		Long:  "Generate markdown documentation for the CLI and save it to the specified output directory.",
+		Args:  cobra.ExactArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			outputDir := args[0]
+			if err := generateMarkdownDocs(cmd.Parent(), outputDir); err != nil {
+				logrus.WithError(err).Fatal("Failed to generate markdown documentation")
```
[the remark this answers is not in the document]
Comment (derekbit) [reply cmd/remote/longhornctl.go]: Updated. Thank you.

**Review by c3y1huang — APPROVED** [review]

**Comment by derekbit** [comment]
@mergify backport v1.7.x

**Pull request merged**